packages/engine/lib/memory/src/shared_memory/segment.rs

```diff
@@ -113,10 +100,23 @@ impl fmt::Display for MemoryId {
     }
 }
 
-impl Borrow<Uuid> for &MemoryId {
-    fn borrow(&self) -> &Uuid {
-        &self.id
+/// Clean up generated shared memory segments associated with a given `MemoryId`.
+pub fn clean_up_by_base_id<Id: Borrow<Uuid>>(id: Id) -> Result<()> {
```

Comment:
I didn't pay attention before, but can we agree on `cleanup` rather than `clean_up`? Currently it's mixed.

Reply:
Seems to be us vs british spelling, https://grammarist.com/usage/cleanup-clean-up/.
Looks like we have almost all "cleanup" and just this one "clean_up".

Reply:
Renamed in https://github.com/hashintel/hash/pull/494/commits/aed6f525977684f2f80281baa2dc75a03680e8a0

Reply:
Yes, it's probably my fault for naming it `MemoryId::clean_up`. Generally, we use American English. (No one likes the additional letters the British introduce 😉 )